core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -225,27 +225,34 @@ private[zio] final class FiberContext[E, A](
     val left  = fork[EL, A](race.left.asInstanceOf[IO[EL, A]])
     val right = fork[ER, B](race.right.asInstanceOf[IO[ER, B]])
 
-    ZIO
-      .effectAsync[R, E, C] { cb =>
-        val leftRegister = left.register0 {
-          case exit0: Exit.Success[Exit[EL, A]] =>
-            complete[EL, ER, A, B](left, right, race.leftWins, exit0.value, raceIndicator, cb)
-          case exit: Exit.Failure[_] => complete(left, right, race.leftWins, exit, raceIndicator, cb)
-        }
-
-        if (leftRegister ne null)
-          complete(left, right, race.leftWins, leftRegister, raceIndicator, cb)
-        else {
-          val rightRegister = right.register0 {
-            case exit0: Exit.Success[Exit[_, _]] =>
-              complete(right, left, race.rightWins, exit0.value, raceIndicator, cb)
-            case exit: Exit.Failure[_] => complete(right, left, race.rightWins, exit, raceIndicator, cb)
-          }
+    for {
+      lid <- left.id
```

Comment:
This will slow down the implementation quite a lot. Instead, you can just make the `fiberId` accessible (without being wrapped in an effect) by making it a protected val in the `FiberContext` constructor.

Reply:
Good idea. Done.